src/test/kotlin/io/openfuture/state/controller/WalletControllerTest.kt

```diff
@@ -0,0 +1,49 @@
+package io.openfuture.state.controller
+
+import com.nhaarman.mockitokotlin2.given
+import io.openfuture.state.base.ControllerTests
+import io.openfuture.state.service.WalletService
+import io.openfuture.state.util.createDummySaveWalletRequest
+import io.openfuture.state.util.createDummyWallet
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
+import org.springframework.boot.test.mock.mockito.MockBean
+
+@WebFluxTest(WalletController::class, ExceptionHandler::class)
+class WalletControllerTest : ControllerTests() {
+
+    @MockBean
+    private lateinit var walletService: WalletService
+
+    @Test
+    fun findByAddressShouldReturnWallet() = runBlocking<Unit> {
+        val wallet = createDummyWallet()
+        val walletDto = WalletController.WalletDto(wallet)
+        given(walletService.findByAddress("this-is-address")).willReturn(wallet)
+
+        webClient.get()
+                .uri("/api/wallets/address/this-is-address")
+                .exchange()
+                .expectStatus().isOk
+                .expectBody()
+                .json(objectMapper.writeValueAsString(walletDto))
```

Comment:
You should not programmatically create JSONs. Instead, you need to have them as static files or inline json right here.
https://github.com/ruslanys/ifunny/blob/master/src/test/kotlin/me/ruslanys/ifunny/controller/api/FeedApiControllerTests.kt